File: scanner/corpora/java/positive/hmac.java
import javax.crypto.Mac;
import javax.crypto.spec.SecretKeySpec;
import java.security.MessageDigest;

public class HmacExample {
    public static void main(String[] args) throws Exception {
        byte[] key = "my-secret-key".getBytes();
        byte[] message = "important data".getBytes();

        Mac mac = Mac.getInstance("HmacSHA256");
        SecretKeySpec keySpec = new SecretKeySpec(key, "HmacSHA256");
        mac.init(keySpec);
        byte[] digest = mac.doFinal(message);
        System.out.println(javax.xml.bind.DatatypeConverter.printHexBinary(digest));
    }
}

File: scanner/corpora/go/positive/hmac.go
package main

import (
	"crypto/hmac"
	"crypto/sha256"
	"fmt"
)

func hmacSign(key, message []byte) []byte {
	h := hmac.New(sha256.New, key)
	h.Write(message)
	return h.Sum(nil)
}

func hmacVerify(key, message, expectedMAC []byte) bool {
	h := hmac.New(sha256.New, key)
	h.Write(message)
	return hmac.Equal(h.Sum(nil), expectedMAC)
}

func main() {
	key := []byte("my-secret-key")
	msg := []byte("important data")
	mac := hmacSign(key, msg)
	valid := hmacVerify(key, msg, mac)
	fmt.Printf("HMAC valid: %v\n", valid)
}
